src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -53,8 +53,17 @@ export class NotebookLinkHandler {
 			this.isMarkdown = this._link.attributes['is-markdown']?.nodeValue === 'true' ? true : false;
 			this.isEncoded = this._link.attributes['is-encoded']?.nodeValue === 'true' ? true : false;
 			this._isFile = this._link.protocol === `${Schemas.file}:` || this._link.protocol === `${Schemas.vscodeFileResource}:`;
-			// Given an anchor element for windows href link will need to use nodeValue instead as that does not encode the url
-			if (isWindows) {
+			// When editing on richtext, for local files paths use pathname to get the filepath of the link
+			// since the nodeValue results are ambiguous depending on OS ex: '/PathToParentFolder/ ./fiename.ipynb'.
+			if (this._isFile) {
+				// check if nodevalue is valid
+				if (notebookUtils.isFilepathValid(this._link.attributes['href']?.nodeValue)) {
+					this._href = this._link.attributes['href']?.nodeValue;
+				} else {
+					this._href = this._link.pathname;
+				}
+			} else if (isWindows) {
```

Comment:
You don't still need to do this logic for windows file paths?

Reply:
so the else part is for web links not file and is still needed.

Reply:
Would be good to add a comment above the else indicating that this would be for web links while the above would be for file paths just to be clear.

Reply:
I would even suggest to have an outer if with the isFile check and then in the else put the previous isWindows/else logic. Makes it easier to follow that those two blocks are what should happen for web links.